src/firefly/java/edu/caltech/ipac/firefly/server/ServCommand.java

```diff
@@ -22,49 +22,39 @@ public void processRequest(HttpServletRequest req, HttpServletResponse res, SrvP
         if (doJsonp) res.setContentType("application/javascript");
 
         String jsonData;
+        JSONObject json=new JSONObject();
+
         try {
             String result = doCommand(new SrvParam(sp.getParamMap()));
 
             if (getCanCreateJson()) {
                 jsonData = result;
             } else {
-                jsonData = "[{" +
-                        "\"success\" :  \"" + true + "\"," +
-                        "\"data\" :  \"" + StringUtils.escapeQuotes(result) + "\"" +
-                        "}]";
-            }
-
-            StringBuilder sb= new StringBuilder(jsonData.length()+10);
-            for(char c : jsonData.toCharArray()) {
-                if (!Character.isISOControl(c) && c>31 && c<127 ) {
-                    sb.append(c);
-                }
+                json.put("success","true" );
+                json.put("data", result );
+                jsonData = "["+json.toJSONString()+"]";
```

Comment:
Using simple Jon is clearly cleaner. On this  line you should not build the array yourself, just simple json do that too.